plugins/lsp/language-servers/default.nix

```diff
@@ -361,6 +368,86 @@ with lib; let
         };
       };
     }
+    {
+      name = "omnisharp";
+      description = "Enable omnisharp language server, for C#";
+      package = pkgs.omnisharp-roslyn;
+      cmd = cfg: [
+        "dotnet"
+        "${cfg.package}/lib/omnisharp-roslyn/OmniSharp.dll"
+      ];
+      settings = cfg: {omnisharp = cfg;};
+      settingsOptions = {
+        enableEditorConfigSupport = mkOption {
+          type = types.nullOr types.bool;
+          default = true;
```

Comment:
Not sure if if it's necessary or not to do this. @GaetanLepage can probably tell.
But we do have helper functions for these like `helpers.defaultNullOpts.mkBool false "some description";`
You can probably look into the `nixd` server declaration just above yours. (You can do this for all the options)

Reply:
Indeed you are right @Tanish2002